### ISSUE TITLE
RFE: Cancel Action run when the newer request was received; Consume fewer GitHub/TF resources

### ISSUE THREAD

**Issue by jamacku** [sclorg/testing-farm-as-github-action#119]
### Version

`main`

### Description

Add a mechanism to cancel existing workflow runs, and TF requests when a new workflow is triggered.

Be more cautious about TF and GitHub resources.

### Reproducer

When a new workflow run is requested (push, comment, etc.), there are two workflows with the `testing-farm-as-github-action` request running at the same time.

This might be an issue for larger projects since there is a limit for a number of running workflows at the same time.

**Comment by obabec** [comment]
This feature would be really helpful for us. We are triggering a lot of jobs for our project https://github.com/debezium/debezium, currently, we are using Pack-it so we can control how many jobs are we triggering. With this, we could fully automate our triggering as older jobs would be canceled.

**Comment by jamacku** [comment]
@obabec I think there is an answer for the GitHub part of your request:

[How to cancel previous runs in the PR when you push new commits(update the current) in github-actions](https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre)

TL;DR, use:

```yml
concurrency: 
  group: ${{ github.head_ref }}
  cancel-in-progress: true
```

**Comment by obabec** [comment]
@jamacku Well, that solves only one part of the issue. You will cancel the GitHub action run/job but you won't actually stop the Testing Farm job in the background. That will still continue...

**Comment by jamacku** [comment]
I know, I have assumed that that's more urgent for you and easier to fix on your side. We are planning to add support for canceling TF requests.

**Comment by jamacku** [comment]
OK, the solution should be pretty straightforward. Assuming that users have `concurrency` enabled in their workflows. Then we add [`post-if: cancelled()`](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runspost-if) script that would cancel the job on the TF side.
